src/backend/src/services/slack.services.ts

```diff
@@ -0,0 +1,289 @@
+              .filter((userSettings) => {
+                return slackIds.some((slackId) => slackId === userSettings.slackId);
+              })
+              .map((user) => user.userId);
+          default:
+            return [];
+        }
+      case 'user':
+        return usersSettings
+          .filter((userSettings) => userSettings.slackId === block.user_id)
+          .map((userSettings) => userSettings.userId);
+      default:
+        //only broadcasts and specific user mentions add recievers to announcements
+        return [];
+    }
+  }
+
+  /**
+   * Given a slack event representing a message in a channel,
+   * make the appropriate announcement change in prisma.
+   * @param event the slack event that will be processed
+   * @param organizationId the id of the organization represented by the slack api
+   * @returns an annoucement if an announcement was processed and created/modified/deleted
+   */
+  static async processMessageSent(event: SlackMessageEvent, organizationId: string): Promise<Announcement | undefined> {
+    let slackChannelName: string;
+    //get the name of the channel from the slack api
+    try {
+      slackChannelName = await getChannelName(event.channel);
+    } catch (error) {
+      slackChannelName = `Unknown_Channel:${event.channel}`;
+    }
+    const dateCreated = new Date(Number(event.event_ts));
+
+    //get the message that will be processed either as the event or within a subtype
+    let eventMessage: SlackMessage;
+
+    if (event.subtype) {
+      switch (event.subtype) {
+        case 'message_deleted':
+          //delete the message using the client_msg_id
+          eventMessage = (event as SlackDeletedMessage).previous_message;
+          try {
+            return AnnouncementService.deleteAnnouncement(eventMessage.client_msg_id, organizationId);
+          } catch (ignored) {
+            return;
+          }
+        case 'message_changed':
+          eventMessage = (event as SlackUpdatedMessage).message;
+          break;
+        default:
+          //other events that do not effect announcements
+          return;
+      }
+    } else {
+      eventMessage = event as SlackMessage;
+    }
+
+    //loop through the blocks of the meta data while accumulating the
+    //text and users notified
+    let messageText = '';
+    let userIdsToNotify: string[] = [];
+
+    //Get the settings of all users in this organization to compare slack ids
+    const users = await UsersService.getAllUsers();
+    const userSettings = await Promise.all(
+      users.map((user) => {
+        return UsersService.getUserSettings(user.userId);
+      })
+    );
+
+    //get the name of the user that sent the message from slack
+    let userName: string = '';
+    try {
+      userName = (await getUserName(eventMessage.user)) ?? '';
+    } catch (ignored) {}
+
+    //if slack could not produce the name of the user, look for their name in prisma
+    if (!userName) {
+      const userIdList = userSettings
```

Comment:
you can replace this with a user.findUnique({ where: { userSettings: { slackId: eventMessage.user } } }) I think.